chrome/content/zotero/xpcom/attachments.js

```diff
@@ -349,6 +349,39 @@ Zotero.Attachments = new function(){
 		}
 		return attachmentItem;
 	});
+
+
+	this.hiddenBrowserImport = function (url, cookieSandbox, importFunction) {
```

Comment:
I'm not sure this belongs here — it's not really about attachments or importing. This is really just a version of loadDocuments() that enforces a 2xx response code, right? Why don't we just put that in loadDocuments itself, and have it fail (so, throw Zotero.HTTP.UnexpectedStatusException (with a fake `xmlhttp` that has a status) or call onError()) instead of calling the processor if there's a non-2xx code? I doubt it's ever going to be necessary or desirable to process an error page.

Reply:
Sounds good. This was mostly because I felt bad about copying so much code from importFromURL, but didn't have the gumption to refactor that and did not explore `loadDocuments`. Thanks.